zebrad/tests/common/get_block_template_rpcs/get_block_template.rs

```diff
@@ -112,7 +138,96 @@ pub(crate) async fn run() -> Result<()> {
     // [Note on port conflict](#Note on port conflict)
     output
         .assert_was_killed()
-        .wrap_err("Possible port conflict. Are there other acceptance tests running?")?;
+        .wrap_err("Possible port conflict. Are there other acceptance tests running?")
+}
+
+/// Accepts an [`RPCRequestClient`], calls getblocktemplate in template mode,
+/// deserializes and transforms the block template in the response into block proposal data,
+/// then calls getblocktemplate RPC in proposal mode with the serialized and hex-encoded data.
+///
+/// Returns an error if it fails to transform template to block proposal or serialize the block proposal
+/// Returns `Ok(())` if the block proposal is valid or an error with the reject-reason if the result is
+/// an `ErrorResponse`.
+///
+/// ## Panics
+///
+/// If an RPC call returns a failure
+/// If the response result cannot be deserialized to `GetBlockTemplate` in 'template' mode
+/// or `ProposalResponse` in 'proposal' mode.
+async fn try_validate_block_template(client: &RPCRequestClient) -> Result<()> {
+    let response_json_result = client
+        .json_result_from_call("getblocktemplate", "[]".to_string())
+        .await
+        .expect("response should be success output with with a serialized `GetBlockTemplate`");
+
+    tracing::info!(
+        ?response_json_result,
+        "got getblocktemplate response, hopefully with transactions"
+    );
+
+    // Propose a new block with an empty solution and nonce field
+    tracing::info!("calling getblocktemplate with a block proposal...",);
+
+    let raw_proposal_block =
+        hex::encode(proposal_block_from_template(response_json_result)?.zcash_serialize_to_vec()?);
+
+    let json_result = client
+        .json_result_from_call(
+            "getblocktemplate",
+            format!(r#"[{{"mode":"proposal","data":"{raw_proposal_block}"}}]"#),
+        )
+        .await
+        .expect("response should be success output with with a serialized `ProposalResponse`");
+
+    tracing::info!(?json_result, "got getblocktemplate proposal response");
+
+    match json_result {
+        ProposalResponse::Valid => Ok(()),
+        ProposalResponse::ErrorResponse { reject_reason, .. } => Err(eyre!(
+            "unsuccessful block proposal validation, reason: {reject_reason:?}"
+        )),
+    }
+}
+
+/// Make a block proposal from [`GetBlockTemplate`]
+fn proposal_block_from_template(
```

Comment:
I think I'll need this function as a separate binary for PR #5944, doing it in a shell script with `jq` is difficult.

I can make these changes in my PR.